jwql/instrument_monitors/common_monitors/bad_pixel_monitor.py

```diff
@@ -0,0 +1,1066 @@
+
+        Returns
+        -------
+        parameters : dict
+            Dictionary of parameters, in the format expected by CRDS
+        """
+        parameters = {}
+        parameters['INSTRUME'] = self.instrument.upper()
+        parameters['SUBARRAY'] = 'FULL'
+        parameters['DATE-OBS'] = datetime.date.today().isoformat()
+        current_date = datetime.datetime.now()
+        parameters['TIME-OBS'] = current_date.time().isoformat()
+        parameters['DETECTOR'] = self.detector.upper()
+        if self.instrument.upper() == 'NIRCAM':
+            if parameters['DETECTOR'] in ['NRCALONG', 'NRCBLONG']:
+                parameters['CHANNEL'] = 'LONG'
+            else:
+                parameters['CHANNEL'] = 'SHORT'
+        return parameters
+
+    def process(self, illuminated_raw_files, illuminated_slope_files, dark_raw_files, dark_slope_files):
+        """The main method for processing darks.  See module docstrings
+        for further details.
+
+        Parameters
+        ----------
+        illuminated_raw_files : list
+            List of filenames (including full paths) of raw (uncal) flat
+            field files. These should all be for the same detector and
+            aperture.
+
+        illuminated_slope_files : list
+            List of filenames (including full paths) of flat field slope
+            files. These should all be for the same detector and aperture
+            and correspond one-to-one with ``illuminated_raw_files``. For
+            cases where a raw file exists but no slope file, the slope
+            file should be None
+
+        dark_raw_files : list
+            List of filenames (including full paths) of raw (uncal) dark
+            files. These should all be for the same detector and
+            aperture.
+
+        dark_slope_files : list
+            List of filenames (including full paths) of dark current slope
+            files. These should all be for the same detector and aperture
+            and correspond one-to-one with ``dark_raw_files``. For
+            cases where a raw file exists but no slope file, the slope
+            file should be None
+        """
+        # Illuminated files - run entirety of calwebb_detector1 for uncal
+        # files where corresponding rate file is 'None'
+        all_files = []
+        badpix_types = []
+        badpix_types_from_flats = ['DEAD', 'LOW_QE', 'OPEN', 'ADJ_OPEN']
+        badpix_types_from_darks = ['HOT', 'RC', 'OTHER_BAD_PIXEL', 'TELEGRAPH']
+        illuminated_obstimes = []
+        if illuminated_raw_files is not None:
+            index = 0
+            badpix_types.extend(badpix_types_from_flats)
+            for uncal_file, rate_file in zip(illuminated_raw_files, illuminated_slope_files):
+                self.get_metadata(uncal_file)
+                if rate_file == 'None':
+                    jump_output, rate_output, junk = pipeline_tools.calwebb_detector1_save_jump(uncal_file, self.data_dir,
```

Comment:
I often see people use `_` for variable names that are not going to be used, it might be a standard way of dealing with that.

Reply:
Done